build.gradle

```diff
@@ -979,3 +983,39 @@ apply from: 'baseline.gradle'
 apply from: 'deploy.gradle'
 apply from: 'tasks.gradle'
 
+project(':iceberg-bom') {
+  apply plugin: 'java-platform'
+
+  dependencies {
+    constraints {
+      def sparkScalaPattern = ~"(.*)-([0-9][.][0-9]+)_([0-9][.][0-9]+)"
```

Comment:
I think it would be good to add a comment describing why were doing some manual work for Spark here, as it's not obvious to everyone